.github/workflows/android.yml

```
@@ -7,32 +7,79 @@ on:
+      - name: Checkout
+        uses: actions/checkout@v4
+      
       - name: set up JDK 21
         uses: actions/setup-java@v4
         with:
           java-version: '21'
           distribution: 'temurin'
           cache: gradle
+      
+      - name: Cache Gradle dependencies
+        uses: actions/cache@v3
```

actionlint (1.7.7)

the runner of "actions/cache@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)